metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -524,8 +525,23 @@ object MetalsEnrichments
       Files.delete(path.dealias.toNIO)
     }
 
+    // This method used to fail on both `.toList` and `delete`
+    // so we want to be extra careful and check if file exists on every step
     def deleteRecursively(): Unit = {
-      path.listRecursive.toList.reverse.foreach(f => if (f.exists) f.delete())
+      path.listRecursive
+        .filter(_.exists)
```

Comment:
I know it looks weird to check if file exists both here and in `deleteIfExits`, but I saw logs where this method was failing already on `toList`, so maybe it's best to be extra careful